source/Ship.cpp

```diff
@@ -3708,7 +3708,31 @@ void Ship::DoGeneration()
 	double maxHull = attributes.Get("hull");
 	hull = min(hull, maxHull);
 
-	isDisabled = isOverheated || hull < MinimumHull() || (!crew && RequiredCrew());
+	bool isIncapacitated = hull < MinimumHull() || (!crew && RequiredCrew());
+	isDisabled = isOverheated || isIncapacitated;
+
+	if(!isIncapacitated)
+	{
+		double coolingEfficiency = CoolingEfficiency();
+		heat -= coolingEfficiency * attributes.Get("cooling");
+		double activeCooling = coolingEfficiency * attributes.Get("active cooling");
+		// Apply active cooling. The fraction of full cooling to apply equals
+		// your ship's current fraction of its maximum temperature.
+		if(activeCooling > 0. && heat > 0. && energy >= 0.)
+		{
+			// Handle the case where "active cooling"
+			// does not require any energy.
+			double coolingEnergy = attributes.Get("cooling energy");
+			if(coolingEnergy)
+			{
+				double spentEnergy = min(energy, coolingEnergy * min(1., Heat()));
```

Comment:
Shouldnt it just always be at full power if you:re overheated and its active cooling?
It makes no sense to make it run at 0.9 in that case I think, even for a short time, and it'd save us some calculations + it would buff active cooling a bit, which needs it

Reply:
If you're overheated it will always be full power, because your heat will be 100%. I could optimise the code by having it not bother checking heat if you're overheated, which was how I originally had it, but quyykk suggested that it would be better to rearrange it so that there aren't two seperate versions of the same code.

I don't see where you're getting the 0.9 figure from - as I understand it, this code will cause it to scale between 0 and 100%, reaching 100% at 100% heat. I do think it'd be better if it reached 100% cooling a little before that, but that would be a separate change and should be a separate PR.

Reply:
I mean if quyykk asked it this way then do ignore me - but I feel like he meant to make the two codes be the same function then?
Also I think that when overheated you stay that way until reaching below 90%, from looking at the bar when it happens to me, but mb I'm wrong there

Reply:
>  Also I think that when overheated you stay that way until reaching below 90%, from looking at the bar when it happens to me, but mb I'm wrong there

Yeah, the code for that is a couple lines above. You make a good point though, so I agree. 

> but I feel like he meant to make the two codes be the same function then?

I mean, you could do that but there would be a lot of variables to pass so idk, it doesn't really seem worth it, especially since it would not contain a lot of code.

Reply:
Ok, I misunderstood what Hurleveur meant - yes, I agree active cooling should run at 100% in the case where you're below 100% heat but still overheated. I have updated the code accordingly.

Reply:
Glad we agree, if we're not merging the code then I dont see why it wouldnt be context-dependant

Reply:
> > but I feel like he meant to make the two codes be the same function then?
> 
> 
> 
> I mean, you could do that but there would be a lot of variables to pass so idk, it doesn't really seem worth it, especially since it would not contain a lot of code.

In that case, how about a lambda? You can just capture all in a lambda.